app/src/main/java/com/arygm/quickfix/model/profile/Profile.kt

```diff
@@ -3,18 +3,43 @@ package com.arygm.quickfix.model.profile
 import com.google.firebase.Timestamp
 import com.google.firebase.firestore.GeoPoint
 
-data class Profile(
+open class Profile(
     val uid: String,
     val firstName: String,
     val lastName: String,
     val email: String,
     val birthDate: Timestamp,
     val description: String,
+    val location: GeoPoint? = null
+)
+
+class UserProfile(
     val isWorker: Boolean = false,
+    uid: String,
+    firstName: String,
+    lastName: String,
+    email: String,
+    birthDate: Timestamp,
+    description: String,
+    location: GeoPoint? = null
```

Comment:
Considering that a user may have multiple favorite locations, it would be better to store these as a list of `Location`. Regarding the type, we agreed to follow a similar setup to the one used in the bootcamp, so I think creating a `Location` data class would be the best approach.

Reply:
I think it should be a map of location name to Geopoint (`Map<String, GeoPoint>`). We should not Use `Location` in the model since it is neither a basic type nor a firestore specific type (we can't store types that are not basic or firestore specific)

Reply:
I meant creating a Location data class as the one in the bootcamp (as we discussed in the last meeting). I’m not referring to a specific type Location. We can convert it to a Map directly in the repository, but doing it beforehand is also fine. However, I still believe that we should create a Location class. And also let it as Location not a Map since the logic of converting it to a Map is specific to the repo.